controllers/argocd/secret/secret.go

```diff
@@ -17,7 +39,494 @@ type SecretReconciler struct {
+			return err
+		}
+
+		if err = controllerutil.SetControllerReference(sr.Instance, argocdSecret, sr.Scheme); err != nil {
+			sr.Logger.Error(err, "reconcileClusterPermissionsSecret: failed to set owner reference for secret", "name", argocdSecret.Name, "namespace", argocdSecret.Namespace)
+		}
+
+		if err = workloads.CreateSecret(argocdSecret, sr.Client); err != nil {
+			sr.Logger.Error(err, "reconcileClusterPermissionsSecret: failed to create secret", "name", argocdSecret.Name, "namespace", argocdSecret.Namespace)
+			return err
+		}
+		sr.Logger.V(0).Info("reconcileClusterPermissionsSecret: secret created", "name", argocdSecret.Name, "namespace", argocdSecret.Namespace)
+		return nil
+	}
+
+	argocdSecretChanged := false
+	if existingArgoCDSecret.Data == nil {
+		existingArgoCDSecret.Data = make(map[string][]byte)
+	}
+	if existingArgoCDSecret.Data[common.ArgoCDKeyServerSecretKey] == nil {
+		sessionKey, err := argocdcommon.GenerateArgoServerSessionKey()
+		if err != nil {
+			sr.Logger.Error(err, "reconcileArgoCDSecret: failed to generate session key")
+			return err
+		}
+		existingArgoCDSecret.Data[common.ArgoCDKeyServerSecretKey] = sessionKey
+		argocdSecretChanged = true
+	}
+
+	if ArgoAdminPasswordChanged(existingArgoCDSecret, credsSecret) {
+		sr.Logger.V(1).Info("admin password changed, updating secret", "name", existingArgoCDSecret.Name, "namespace", existingArgoCDSecret.Namespace)
+		pwBytes, ok := credsSecret.Data[common.ArgoCDKeyAdminPassword]
+		if ok {
+			hashedPassword, err := argopass.HashPassword(strings.TrimRight(string(pwBytes), "\n"))
+			if err != nil {
+				sr.Logger.Error(err, "reconcileArgoCDSecret: failed to hash admin credentials")
+				return err
+			}
+			existingArgoCDSecret.Data[common.ArgoCDKeyAdminPassword] = []byte(hashedPassword)
+			existingArgoCDSecret.Data[common.ArgoCDKeyAdminPasswordMTime] = util.NowBytes()
+			argocdSecretChanged = true
+		}
+	}
+
+	if ArgoTLSChanged(existingArgoCDSecret, tlsSecret) {
+		sr.Logger.V(1).Info("TLS key or certificate changed, updating secret", "name", existingArgoCDSecret.Name, "namespace", existingArgoCDSecret.Namespace)
+		existingArgoCDSecret.Data[corev1.TLSCertKey] = tlsSecret.Data[corev1.TLSCertKey]
+		existingArgoCDSecret.Data[corev1.TLSPrivateKeyKey] = tlsSecret.Data[corev1.TLSPrivateKeyKey]
+		argocdSecretChanged = true
+	}
+
+	// TO DO: Let dex controller populate dex oauth client secret information instead of doing it here
+
+	if argocdSecretChanged {
+		if err = workloads.UpdateSecret(existingArgoCDSecret, sr.Client); err != nil {
+			sr.Logger.Error(err, "reconcileArgoCDSecret: failed to update secret", "name", existingArgoCDSecret.Name, "namespace", existingArgoCDSecret.Namespace)
+			return err
+		}
+		sr.Logger.V(0).Info("reconcileArgoCDSecret: secret updated", "name", existingArgoCDSecret.Name, "namespace", existingArgoCDSecret.Namespace)
+		return nil
+	}
 
-	// controller logic goes here
+	// nothing to do
```

Comment:
Do we need this comment? If it's necessary can you add a condition like in which case nothing to do? Just to make it clearer.